SimG4CMS/Forward/src/ZdcSD.cc

```diff
@@ -60,245 +62,337 @@ ZdcSD::ZdcSD(const std::string& name,
   }
 }
 
-void ZdcSD::initRun() { hits.clear(); }
-
-bool ZdcSD::getFromLibrary(const G4Step* aStep) {
-  bool ok = true;
-
-  auto const preStepPoint = aStep->GetPreStepPoint();
-
-  double etrack = preStepPoint->GetKineticEnergy();
-  int primaryID = setTrackID(aStep);
-
+void ZdcSD::initRun() {
+  if (useShowerLibrary) {
+    G4ParticleTable* theParticleTable = G4ParticleTable::GetParticleTable();
+    showerLibrary->initRun(theParticleTable);
+  }
   hits.clear();
+}
 
-  // Reset entry point for new primary
-  resetForNewPrimary(aStep);
-
-  if (etrack >= zdcHitEnergyCut) {
-    // create hits only if above threshold
+bool ZdcSD::ProcessHits(G4Step* aStep, G4TouchableHistory*) {
+  NaNTrap(aStep);
 
+  if (aStep == nullptr) {
+    return true;
+  } else {
+    if (useShowerLibrary) {
```

Comment:
Maybe remove the `else` branch, since it spans the whole function body? And please leave a note why the commented out section

```
    if (useShowerLibrary) {
      getFromLibrary(aStep);
    }
```

was left.